core/benches/hash.rs

```diff
@@ -100,12 +99,15 @@ fn criterion_benchmark(c: &mut Criterion) {
         },
     );
     group.bench_function(BenchmarkId::new("xxhash64", BATCH_SIZE), |b| {
-        let input = vec![a3.clone(), a4.clone()];
-        let mut dst = vec![0; BATCH_SIZE];
+        let inputs = &[
+            ColumnarValue::Array(a3.clone()),
+            ColumnarValue::Array(a4.clone()),
+            ColumnarValue::Scalar(ScalarValue::Int64(Some(42i64))),
+        ];
 
         b.iter(|| {
             for _ in 0..NUM_ITER {
-                create_xxhash64_hashes(&input, &mut dst).unwrap();
+                spark_xxhash64(inputs).unwrap();
```

Comment:
is this change related? Looks `spark_xxhash64` covers more code, so performance numbers may change...